src/box/session.cc

```diff
@@ -375,3 +376,72 @@ generic_session_sync(struct session *session)
 	(void) session;
 	return 0;
 }
+
+API_EXPORT int
+box_session_id()
+{
+	return current_session()->id;
+}
+
+API_EXPORT bool
+box_session_exists(uint64_t sid)
+{
+	return session_find(sid) != NULL;
+}
+
+API_EXPORT const char *
+box_session_peer(uint64_t sid, char *dst, size_t size)
+{
+	struct session *session = session_find(sid);
+	if (session == NULL) {
+		diag_set(ClientError, ER_NO_SUCH_SESSION);
+		return NULL;
+	}
+
+	int fd = session_fd(session);
+	if (fd < 0) {
+		return NULL;
+	}
+
+	struct sockaddr_storage addr;
+	socklen_t addrlen = sizeof(addr);
+	if (sio_getpeername(fd, (struct sockaddr *)&addr, &addrlen) < 0)
+		return NULL;
+
+	const char *tmp = sio_strfaddr((struct sockaddr *)&addr, addrlen);
+	if (strlen(tmp) + 1 < size) {
+		diag_set(ClientError, ER_ILLEGAL_PARAMS, "size is too small");
+		return NULL;
+	}
+	strcpy(dst, tmp);
+
+	return tmp;
+}
+
+API_EXPORT const char *
+box_session_user()
+{
+	struct user *user = user_by_id(current_session()->credentials.uid);
+	if (user)
```

Comment:
For non-boolean values, please, use explicit `==` and `!=`. Here it would be `user != NULL`.